### PR TITLE
Keywords

### PR DESCRIPTION
**Pull request by czhanacek**
Addresses #31

### CODE REVIEW

File: app/controllers/survey_controller.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def create_question
         q.save!
       end
     end
-    @question.save!
+    @question.save(:validate => false)
```
Comment (t-walker) [inline app/controllers/survey_controller.rb]: Why is this here?

Comment (t-walker) [inline app/controllers/survey_controller.rb]: I don't remember much about that flag, so I'm just curious what it's doing.

File: app/controllers/survey_controller.rb [hunk 2]
```diff
@@ -38,11 +38,23 @@ def create_question
       nextOrder += 1
       @question.answers.create(answer_title: params[:answer4], position: nextOrder)
     end
+    @question.save!
     flash[:success] = "Question created successfully!"
```
Comment (t-walker) [inline app/controllers/survey_controller.rb]: Awesome use of the Rails framework.

File: app/views/survey/_question-answer.html.erb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <li>
-  <input type="radio" name="question-<%= questId %>">
-    <%= answerTitle %>
-  </input>
+
+  <%= radio_button_tag "responses[][answer" + questId.to_s + "]", :answer %>
```
Comment (t-walker) [inline app/views/survey/_question-answer.html.erb]: does responses[] need to have an index?

File: test/controllers/survey_controller_test.rb [hunk 4]
```diff
@@ -10,4 +10,46 @@ class ManageSurveyControllerTest < ActionDispatch::IntegrationTest
     get survey_manage_url
     assert_response :success
   end
+
```
Comment (t-walker) [inline test/controllers/survey_controller_test.rb]: Yasssss soo many tests. I love it.

File: app/controllers/survey_controller.rb (hunk 1 above)
Comment (czhanacek) [inline app/controllers/survey_controller.rb]: It prevents the validation checks from happening until all parts of the question have been assembled (and then it runs the validation checks).

File: app/views/survey/_question-answer.html.erb (hunk 3 above)
Comment (czhanacek) [inline app/views/survey/_question-answer.html.erb]: I found this in a guide online and I never got a chance to test if it actually works, but the use case seemed right. It's supposed to build the array so that it has the structure of a Responses object. Again, not 100% sure if it actually works. Maybe @seattlecoder9 has some insight?
